### DISCUSSION TITLE
Change Stoploss after a specific time

### DISCUSSION THREAD

**Discussion by iimka** [kernc/backtesting.py#869] [Q&A] [answered]
I know how to code a trailing stoploss, for example:
```
for trade in self.trades:
   if trade.is_short:
      trade.sl = min (trade.sl or +np.inf, self.data.Low * 1.02)
```
But how can I change the stoploss to the entry price after a specific time? For the below example, black line is my expected stoploss.
![2023-01-05_15-57-10](https://user-images.githubusercontent.com/55306556/210731112-2c7200e2-7093-41de-b3fb-25d92f730ca9.jpg)

**Comment by kernc** [comment, marked as answer]
Look into [`Trade.entry_bar`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.entry_bar) (compare it with `len(self.data.index)`) and [`Trade.entry_price`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Trade.entry_price)?

**Comment by iimka** [comment]
Yes, it is what I want!
`
for trade in self.trades: 
print(len(self.data.index))
print(trade.entry_bar)
print(trade.entry_price)
`

![image](https://user-images.githubusercontent.com/55306556/210916575-0761d147-5366-4f57-8ae1-dcfc4770499e.png)
